tests/unit_tests/job_queue/test_lsf_driver.py

```diff
@@ -75,75 +121,216 @@ def mock_bjobs(tmp_path):
     os.chmod(script_path, 0o755)
 
 
-def make_mock_bsub(script_path):
-    script_path.write_text(
-        "#!/usr/bin/env python3"
-        + dedent(
-            """
-           import random
-           import subprocess
-           import sys
-
-           job_dispatch_path = sys.argv[-2]
-           run_path = sys.argv[-1]
-
-           # Write a file with the runpaths to where the jobs are running and
-           # writing information we later need when providing statuses for the
-           # jobs through the mocked bjobs command
-           with open("job_paths", "a+", encoding="utf-8") as jobs_file:
-               jobs_file.write(f"{run_path}\\n")
-
-           # Just a log for testpurposes showing what is thrown against the
-           # bsub command
-           with open("bsub_log", "a+", encoding="utf-8") as f:
-               f.write(f"{str(sys.argv)}\\n")
-
-           # Assigning a "unique" job id for each submitted job and print. This
-           # is how LSF provide response to ERT with the ID of the job.
-           _id = str(random.randint(0, 10000000))
-           print(f"Job <{_id}> is submitted to default queue <normal>.")
-
-           # Launch job-dispatch
-           subprocess.Popen([job_dispatch_path, run_path])
-           """
-        )
-    )
-    os.chmod(script_path, 0o755)
+class MockRealizationState:
+    def verify_kill(self):
+        print("Realization was verified killed")
 
 
-def make_failing_bsub(script_path, success_script):
-    """
-    Approx 3/10 of the submits will fail due to the random generator in the
-    created mocked bsub script. By using the retry functionality towards
-    queue-errors in job_queue.cpp we should still manage to finalize all our runs
-    before exhausting the limits
-    """
-    script_path.write_text(
-        "#!/usr/bin/env python3"
-        + dedent(
-            f"""
-            import random
-            import sys
-            import subprocess
-
-            num = random.random()
-            if num > 0.7:
-                exit(1)
-            subprocess.call(["python", "{success_script}"] + sys.argv)
-            """
-        )
+def create_fake_bjobs_result(dir: str, job_id: str, status: str):
+    # print("JOBID USER STAT QUEUE FROM_HOST EXEC_HOST JOB_NAME SUBMIT_TIME")
+    Path(dir / "mocked_result").write_text(
+        f"{job_id}\tpytest\t{status}\ttest_queue\thost1\thost2\ttest_job\t{str(datetime.now())}"
     )
 
-    os.chmod(script_path, 0o755)
 
+async def test_submit_failure_script_exit(mock_bsub, caplog, tmpdir, monkeypatch):
+    monkeypatch.chdir(tmpdir)
+    lsf_driver = LSFDriver(None)
+    lsf_driver.set_option("BSUB_CMD", tmpdir / "mock_bsub")
+    lsf_driver._max_attempt = 3
+    lsf_driver._SLEEP_PERIOD = 0.2
```

Comment:
This makes the test spend at least 0.6 seconds to complete? Can you set it to zero as long as you are not explicitly testing that the total time adds up?

Reply:
Yesss! Why didnt I see this before 🤔